packages/block-library/src/paragraph/theme.scss

```diff
@@ -0,0 +1,4 @@
+p {
```

Comment:
Similar question here. 

Because the purpose of this PR is to *remove the intrinsic block margin*, we have to actually re-apply those margins somewhere else. 

Where should these rules live? Should they be in a vanilla editor style only? Should they be opinionated editor styles as they are here? Or something else? Really difficult question, but it gets at the _heart_ of some of the CSS bleed issues this PR also aims to fix. CC: @m-e-h for expert commentary here thank you.

Reply:
Same as with headings. If the theme has their own `p` margins, those will get used. Otherwise it'll fall back to these styles or to the browser styles. 

Both Chrome and Firefox have this:
```
p {
    margin-block-start: 1em;
    margin-block-end: 1em;
}
```
_which is the same as `margin-top` and `margin-bottom`_

Here are some samples of default browser styles for reference: https://github.com/mozdevs/cssremedy/tree/master/process/UA_stylesheets

Reply:
> Where should these rules live?

Good question. Not really sure.

I do think that general selectors like this should probably never go in a block specific stylesheet. 
Sure, this is a "paragraph" block and it makes sense to style a `p` here but it sets a poor example. I wouldn't want to encourage blocks to freely use element selectors in their block styles.

So, the more I think about it, maybe a separate set of styles like this, imported or enqueued before any other block styles, is a good way to go. 
It would be invisible to any theme _(most)_ that had their own styles for these elements.

Reply:
Would it make a difference if you viewed it from a Drupal developer's perspective?

Reply:
This feels appropriate for some kind of "canvas reset" file, which is only enqueued in the backend.

Reply:
> Same as with headings. If the theme has their own p margins, those will get used. Otherwise it'll fall back to these styles or to the browser styles.

Just to clarify, `Otherwise it'll fall back to these styles or to the browser styles.` is not correct for the editing canvas. In the editing canvas it will fall back to the WP-admin styles, which look like this:

![Screenshot 2019-03-15 at 11 16 07](https://user-images.githubusercontent.com/1204802/54424698-c2395a00-4713-11e9-84fa-1c68721a5c07.png)

I'm starting to think that the idea by @getdave and later @chrisvanpatten of a "normalization" stylesheet might be the correct place to put styles like these, which are unfortunately necessary for the time being.